Frontend/student-hub/src/pages/Friends.js

```diff
@@ -0,0 +1,279 @@
+import React, { useState, useEffect } from 'react';
+import { Container, Grid, List, ListItem, ListItemText, ListItemSecondaryAction, IconButton, Typography, Paper, InputBase} from '@mui/material'
+import  Remove  from '@mui/icons-material/Remove'
+import  Add  from '@mui/icons-material/Add'
+import SearchIcon from '@mui/icons-material/Search';
+import { styled } from '@mui/system';
+
+// search styling
+const SearchContainer = styled(Paper)({
+    display: 'flex',
+    alignItems: 'center',
+    padding: '2px 4px',
+    marginBottom: '20px',
+    marginTop: '20px'
+});
+
+const SearchInput = styled(InputBase)({
+    marginLeft: '8px',
+    flex: 1,
+});
+
+const FriendsList = () => {
+    const [friends, setFriends] = useState([]);
+    const [availableFriend, setAvailableFriend] = useState([]);
+    const [recommendedFriends, setRecommendedFriends] = useState([]);
+    const [searchTerm, setSearchTerm] = useState('');
+    const [filteredData, setFilteredData] = useState({
+        friends: [],
+        recommendedFriends: [],
+        availableFriend: []
+    });
+
+    //fetch data
+    useEffect(() => {
+        const fetchData = async () => {
+            await fetchFriends();
+            await fetchAvailableFriend();
+            await fetchRecommendedFriends();
+        };
+        fetchData();
+    }, []);
+
+
+    //initialize filtered data after fetching
+    useEffect(() => {
+    setFilteredData({
+        friends,
+        recommendedFriends,
+        availableFriend
+    });
+}, [friends, recommendedFriends, availableFriend]);
+
+// friends
+const fetchFriends = async () => {
+    try {
+        const userId = sessionStorage.getItem('userId');
+        const token = sessionStorage.getItem('token');
+        const response = await fetch(`http://localhost:8080/friends/${userId}`,{
```

Comment:
similarly I'd recommend a constant string that you can share across your codebase for your API URL. In the case where this would get deployed or change URLs you'll have to do quite a bit of refactoring.